pkg/apis/networking/validation/validation_test.go

```diff
@@ -33,1022 +33,315 @@ import (
-							},
-							{
-								Protocol: &protocolSCTP,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 7777},
-							},
-						},
-					},
-				},
+
+	// Tweaks used below.
+	setIngressEmptyFirstElement := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress = []networking.NetworkPolicyIngressRule{{}}
+	}
+
+	setIngressFromEmptyFirstElement := func(networkPolicy *networking.NetworkPolicy) {
+		if networkPolicy.Spec.Ingress == nil {
+			setIngressEmptyFirstElement(networkPolicy)
+		}
+		networkPolicy.Spec.Ingress[0].From = []networking.NetworkPolicyPeer{{}}
+	}
+
+	setIngressFromIfEmpty := func(networkPolicy *networking.NetworkPolicy) {
+		if networkPolicy.Spec.Ingress == nil {
+			setIngressEmptyFirstElement(networkPolicy)
+		}
+		if networkPolicy.Spec.Ingress[0].From == nil {
+			setIngressFromEmptyFirstElement(networkPolicy)
+		}
+	}
+
+	setIngressEmptyPorts := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress = []networking.NetworkPolicyIngressRule{
+			{
+				Ports: []networking.NetworkPolicyPort{{}},
 			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-					},
-				},
-				Ingress: []networking.NetworkPolicyIngressRule{
-					{
-						From: []networking.NetworkPolicyPeer{
-							{
-								IPBlock: &networking.IPBlock{
-									CIDR:   "fd00:192:168::/48",
-									Except: []string{"fd00:192:168:3::/64", "fd00:192:168:4::/64"},
-								},
+		}
+	}
+
+	setIngressPorts := func(ports ...networking.NetworkPolicyPort) netpolTweak {
+		return func(np *networking.NetworkPolicy) {
+			if np.Spec.Ingress == nil {
+				setIngressEmptyFirstElement(np)
+			}
+			np.Spec.Ingress[0].Ports = make([]networking.NetworkPolicyPort, len(ports))
+			for i, p := range ports {
+				np.Spec.Ingress[0].Ports[i] = p
+			}
+		}
+	}
+
+	setIngressFromPodSelector := func(k, v string) func(*networking.NetworkPolicy) {
+		return func(networkPolicy *networking.NetworkPolicy) {
+			setIngressFromIfEmpty(networkPolicy)
+			networkPolicy.Spec.Ingress = []networking.NetworkPolicyIngressRule{
+				{
+					From: []networking.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{k: v},
 							},
 						},
 					},
 				},
-			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
+			}
+		}
+	}
+
+	setIngressFromNamespaceSelector := func(networkPolicy *networking.NetworkPolicy) {
+		setIngressFromIfEmpty(networkPolicy)
```

Comment:
This one is so much shorter than the equialent `FromPodSelector`?  I'd expect them to be congruent